src/test/ui/const-generics/issues/issue-98813.rs

```diff
@@ -0,0 +1,13 @@
+// build-pass
+
+#![allow(incomplete_features)]
+#![feature(adt_const_params)]
+
+const F: f64 = 12.5;
+
+pub fn func<const F: f64>() {}
+
+fn main() {
+    let _ = func::<14.0>();
```

Comment:
Do we really want to accept this program? Obviously it shouldn't ICE, but IMO it should be rejected.